### PR TITLE
#40 Support multiple policies

### LINKED ISSUE #40: Support multiple policies
Currently only supports a policy, but in many cases, we need a variety of strategies for different current limit, so I would like to modify the code to support multiple policies, you think of this idea? I now have a fork branch that does this    [23mf:feat-support-policies](https://github.com/23mf/spring-cloud-zuul-ratelimit/tree/feat-support-policies)

### PR DESCRIPTION
**Pull request by lchayoun**
Fixes #40

### CODE REVIEW

**Comment by lchayoun** [comment]
@marcosbarbero please take a look
I've added support for multiple policies while keeping the current interface (to be removed at a later version)

I didn't add the support for filtering specific url for a policy since I think it is not the intention of this project.

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java [hunk 1]
```diff
@@ -30,7 +34,12 @@
 import com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.filters.RateLimitPostFilter;
 import com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.filters.RateLimitPreFilter;
 import com.netflix.zuul.ZuulFilter;
-
+import java.util.ArrayList;
+import java.util.Collection;
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java]: MINOR Remove this unused import 'java.util.Collection'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java [hunk 2]
```diff
@@ -45,10 +54,10 @@
 import org.springframework.data.redis.connection.RedisConnectionFactory;
 import org.springframework.data.redis.core.RedisTemplate;
 import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java]: MINOR Remove this unused import 'org.springframework.http.client.ClientHttpRequestInterceptor'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java [hunk 3]
```diff
@@ -30,7 +34,12 @@
 import com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.filters.RateLimitPostFilter;
 import com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.filters.RateLimitPreFilter;
 import com.netflix.zuul.ZuulFilter;
-
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java]: MINOR Remove this unused import 'org.springframework.beans.factory.annotation.Autowired'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java [hunk 4]
```diff
@@ -45,10 +54,10 @@
 import org.springframework.data.redis.connection.RedisConnectionFactory;
 import org.springframework.data.redis.core.RedisTemplate;
 import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.web.client.RestTemplate;
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java]: MINOR Remove this unused import 'org.springframework.web.client.RestTemplate'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by marcosbarbero** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java [hunk 5]
```diff
@@ -134,4 +139,30 @@ public RateLimiter inMemoryRateLimiter() {
...
+            if (defaultPolicy != null) {
+                ArrayList<Policy> defaultPolicies = Lists.newArrayList(defaultPolicy);
+                defaultPolicies.addAll(rateLimitProperties.getDefaultPolicyList());
+                rateLimitProperties.setDefaultPolicyList(defaultPolicies);
+            }
+            rateLimitProperties.getPolicies().forEach((route, policy) ->
+                rateLimitProperties.getPolicyList().compute(route, (k, v) -> getPolicies(policy, v)));
+        }
+
+        private List<Policy> getPolicies(Policy policy, List<Policy> v) {
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/RateLimitAutoConfiguration.java]: maybe a better variable name than `v`?

**Pull request merged**